src/cls/IPM/ResourceProcessor/Artifactory.cls

```diff
@@ -0,0 +1,366 @@
+
+/// Comma-separated list of resource attribute names that this processor uses.
+Parameter ATTRIBUTES As STRING = "InstallDirectory,Filter";
+
+/// Path to which the artifact should be unpacked upon activation; may contain expressions.
+Property InstallDirectory As %String(MAXLEN = "") [ Aliases = {Target,Dest}, Required ];
+
+/// As JSON, the contents of the filter to be used in the items.find() query in JFrog Artifactory
+/// For more details on specific values, visit <a href="https://jfrog.com/help/r/jfrog-artifactory-documentation/artifactory-query-language" target="_blank">JFrog AQL Documentation</a>
+Property Filter As %String(MAXLEN = "");
+
+Method OnAfterPhase(
+	phase As %String,
+	ByRef params) As %Status
+{
+    #dim ex As %Exception.AbstractException
+    set sc = $$$OK
+    try {
+        if (phase = "Activate") {
+            set verbose  = $get(params("Verbose"),0)
+            // If install directory exists, use that
+            // Else, download from artifactory
+            set downloadDirectory = ##class(%File).NormalizeDirectory(..ResourceReference.Module.Root_..ResourceReference.Name)
+            if '##class(%File).DirectoryExists(downloadDirectory) {
+                if (verbose) {
+                    write !, "No artifact found in "_downloadDirectory_", will now download it from artifactory."
+                }
+                do ..DownloadArtifact(.params)
+            } else {
+                if (verbose) {
+                    write !, "Artifact found in "_downloadDirectory_". Will use that instead of downloading a new one."
+                }
+            }
+            // If artifact fails to download in dev mode then the directory won't exist but execution won't error either
+            if ##class(%File).DirectoryExists(downloadDirectory) {
+                // copy everything from /<module root>/<resource name>/ into the final destination
+                // overwrite existing files if there are any in destination dir
+                if (verbose) {
+                    write !, "Going to copy extracted contents from "_downloadDirectory_" to "_..InstallDirectory
+                }
+                // Copy contents over with pDeleteBeforeCopy = 1
+                if '##class(%Library.File).CopyDir(downloadDirectory, ..InstallDirectory,,,1) {
+                    $$$ThrowStatus($$$ERROR($$$GeneralError, "Failed to copy to directory: "_..InstallDirectory))
+                }
+                if (verbose) {
+                    write !, "Copy Succeeded."
+                }
+            }
+        }
+    } catch ex {
+        set sc = ex.AsStatus()
+    }
+    quit sc
+}
+
+Method OnExportItem(
+	fullExportPath As %String,
+	itemName As %String,
+	ByRef itemParams,
+	ByRef params,
+	Output itemHandled As %Boolean = 0) As %Status
+{
+    set sc = $$$OK
+    try {
+        if (params("cmd") = "package") {
```

Comment:
No need for this check (actually you don't want it since this would not export if the publish command is run)